packages/start/src/client/createServerFn.ts

```diff
@@ -365,7 +365,10 @@ const applyMiddleware = (
         context,
         sendContext,
         headers,
-        result: userResult?.result ?? (mCtx as any).result,
+        result:
```

Comment:
I'm not going to lie. I still don't understand why this change fixes the issue. Since at runtime, both conditions should resolve the same thing. 
¯\_(ツ)_/¯

Reply:
previously, when userResult.result was `null`, we retuned mCtx.result because ?? returns the fallback for both null and undefined. 

Now, since we check for undefined explicitly, null will be "passed through".

Reply:
Omg... I totally thought that `??` only ever passed through when it was `undefined` 🤯.